sycl/include/CL/sycl/detail/context_impl.hpp

```diff
@@ -28,55 +27,110 @@ class device;
 
   ~context_impl();
 
+  /// Gets OpenCL interoperability context handle.
+  ///
+  /// @return an instance of OpenCL cl_context.
   cl_context get() const;
 
+  /// Checks if this context is a host context.
+  ///
+  /// @return true if this context is a host context.
   bool is_host() const;
 
-  platform get_platform() const;
-
-  vector_class<device> get_devices() const;
-
+  /// Gets asynchronous exception handler.
+  ///
+  /// @return an instance of SYCL async_handler.
   const async_handler &get_async_handler() const;
 
+  /// Queries this context for information.
+  ///
+  /// The return type depends on information being queried.
   template <info::context param>
   typename info::param_traits<info::context, param>::return_type
   get_info() const;
 
-  // Returns underlying native context object (if any) w/o reference count
-  // modification. Caller must ensure the returned object lives on stack only.
-  // It can also be safely passed to the underlying native runtime API.
-  // Warning. Returned reference will be invalid if context_impl was destroyed.
+  /// Gets the underlying context object (if any) without reference count
+  /// modification.
+  ///
+  /// Caller must ensure the returned object lives on stack only. It can also
+  /// be safely passed to the underlying native runtime API. Warning. Returned
+  /// reference will be invalid if context_impl was destroyed.
+  ///
+  /// @return an instance of raw plug-in context handle.
   RT::PiContext &getHandleRef();
+
+  /// Gets the underlying context object (if any) without reference count
+  /// modification.
+  ///
+  /// Caller must ensure the returned object lives on stack only. It can also
+  /// be safely passed to the underlying native runtime API. Warning. Returned
+  /// reference will be invalid if context_impl was destroyed.
+  ///
+  /// @return an instance of raw plug-in context handle.
   const RT::PiContext &getHandleRef() const;
 
+  /// Gets cached programs.
+  ///
+  /// @return a map of cached programs.
   std::map<OSModuleHandle, RT::PiProgram> &getCachedPrograms() {
-    return m_CachedPrograms;
+    return MCachedPrograms;
   }
+
+  /// Gets cached kernels.
+  ///
+  /// @return a map of cached kernels.
   std::map<RT::PiProgram, std::map<string_class, RT::PiKernel>> &
   getCachedKernels() {
-    return m_CachedKernels;
+    return MCachedKernels;
   }
 
+  /// Gets USM dispatcher.
+  ///
+  /// @return a pointer to USM dispatcher.
   std::shared_ptr<usm::USMDispatcher> getUSMDispatch() const;
 private:
-  async_handler m_AsyncHandler;
-  vector_class<device> m_Devices;
-  RT::PiContext m_Context;
-  platform m_Platform;
-  bool m_OpenCLInterop;
-  bool m_HostContext;
-  std::map<OSModuleHandle, RT::PiProgram> m_CachedPrograms;
-  std::map<RT::PiProgram, std::map<string_class, RT::PiKernel>> m_CachedKernels;
-  std::shared_ptr<usm::USMDispatcher> m_USMDispatch;
+  async_handler MAsyncHandler;
```

Comment:
In LLVM coding style M prefix is not needed for class fields.

Reply:
I agree it is not necessary to have `M` prefix, but I think code is more readable when you can easily distinguish class fields and local variables.